contracts/contracts/core/GasTankManager.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BSL 1.1
+pragma solidity 0.8.26;
+
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {IGasTankManager} from "../interfaces/IGasTankManager.sol";
+import {GasTankManagerStorage} from "./GasTankManagerStorage.sol";
+import {Errors} from "../utils/Errors.sol";
+
+/// @title GasTankManager
+/// @notice Coordinates on-demand ETH bridging so EOA gas tank balances stay funded for mev-commit transactions.
+/// @dev The RPC provider manages EOA gas tank balances on the Mev Commit chain.
+/// @dev Flow overview:
+/// - EOA (L1)
+///   - Authorizes and sets this contract as a delegate against its own EOA address. (ERC-7702)
+///   - Calls `sendMinimumDeposit` to send the initial ETH funds to their gas tank on the Mev Commit chain.
+/// - Provider (mev-commit)
+///   - Triggers a top-up via `topUpGasTank` when the gas tank balance cannot cover the next mev-commit transaction. This transfer amount is always the difference
```

Comment:
This naming is wrong. Again there is no mev-commit concept here. The topup is sent to the operator account of the RPC service.

Reply:
Updated